Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,308 @@
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Default port to connection
+     */
+    const DEFAULT_PORT = 5672;
+
+    /**
+     * Default host to connect
+     */
+    const DEFAULT_HOST = 'localhost';
+
+    /**
+     * @var string
+     */
+    protected $format = AsyncQueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var AMQPStreamConnection
+     */
+    protected $stream;
+
+    public function __destruct()
+    {
+        $this->disconnect();
+    }
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
+     *
+     * {@inheritDoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $urls = explode(',', $host);
+
+        foreach ($urls as $url) {
+            $parsedUrl = parse_url($url);
+
+            $this->connectionsData[] = [
+                'host' => $parsedUrl['host'] ?? self::DEFAULT_HOST,
+                'user' => $username,
+                'password' => $password,
+                'vhost' => $vhost,
+                'port' => $parsedUrl['port'] ?? self::DEFAULT_PORT,
+            ];
+        }
+    }
+
+    /**
+     * Method responsible to open the connection with the broker.
+     *
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     *
+     * @throws \Exception
+     */
+    public function connect($shuffle = true)
+    {
+        $this->shuffleConnections($shuffle);
+        try {
+            $this->stream = AMQPStreamConnection::create_connection($this->connectionsData, []);
+        } catch (AMQPIOException $exception) {
+            throw new \AMQPConnectionException($exception->getMessage());
```

Comment:
Careful here, this is an `ext-amqp` native exception, which is not available without that extension. You might want to replace this with `AMQPProtocolException`.

Reply:
Done. Replaced by the AMQPProtocolConnectionException.